include/class.dept.php

```diff
@@ -205,6 +200,42 @@ function getAvailableMembers() {
         return $this->getMembers(array('available'=>1));
     }
 
+    function getPrimaryMembers() {
+
+        if (!isset($this->_primary_members)) {
+            $members = clone $this->getMembers();
+            $members->filter(array('dept_id' =>$this->getId()));
+            $members = Staff::nsort($members);
+            $this->_primary_members = $members->all()->asArray();
+        }
+
+        return $this->_primary_members;
+    }
+
+    function getExtendedMembers() {
+
+        if (!isset($this->_exended_members)) {
+            // We need a query set so we can sort the names
+            $members = StaffDeptAccess::objects();
+            $members->filter(array('dept_id' => $this->getId()));
+            $members = Staff::nsort($members, 'staff__');
+            $extended = array();
+            foreach($members->all()->asArray() as $member) {
+                if (!$member->staff) continue;
+                // Annoted the staff model with alerts and role
+                $extended[] = new AnnotatedModel($member->staff, array(
```

Comment:
I worked on a variation to the InstrumentedList in the custom queue feature to automate this trick. Perhaps we come up with a nice way to describe it in the meta data without duplicating code